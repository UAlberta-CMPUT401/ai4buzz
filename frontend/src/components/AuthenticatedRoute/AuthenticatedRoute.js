import React from 'react';
import Cookies from 'js-cookie';
import { Redirect, Route } from 'react-router-dom';

const AuthenticatedRoute = (props) => {
  if (Cookies.get('access_token')) {
    return <Route {...props} />;
  }
  return <Redirect to="/login" />;
};

export default AuthenticatedRoute;
