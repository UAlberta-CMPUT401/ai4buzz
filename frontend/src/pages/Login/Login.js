import React, { useState } from 'react';
import Cookies from 'js-cookie';
import { useHistory } from 'react-router-dom';

import { login } from '../../services/auth';
import SigninForm from '../../components/SigninForm/SigninForm';
import styles from './Login.module.css';

const Login = () => {
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();
  const [error, setError] = useState();
  const history = useHistory();

  const handleSubmit = async () => {
    try {
      const res = await login(email, password);
      Cookies.set('access_token', res.data.access_token, { path: '/' });
      history.push('/analyze-image');
    } catch (e) {
      setError(e.response.data);
    }
  };

  return (
    <div className={styles.loginPage}>
      <h1 className={styles.title}>AI4Buzz</h1>
      <SigninForm
        authType="Login"
        onSubmit={handleSubmit}
        onEmailChange={setEmail}
        onPasswordChange={setPassword}
        error={error}
      />
    </div>
  );
};

export default Login;
