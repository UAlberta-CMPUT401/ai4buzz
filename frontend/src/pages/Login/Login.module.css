.loginPage {
  width: 100vw;
  margin-top: 2rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.title {
  font-family: 'Cutive Mono';
  font-size: 3rem;
}
