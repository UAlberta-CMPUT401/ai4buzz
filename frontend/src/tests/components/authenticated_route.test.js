import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import Login from '../../pages/Login/Login';
import AuthenticatedRoute from '../../components/AuthenticatedRoute/AuthenticatedRoute';

it('Renders authenticated route page correctly if access token is available', () => {
  Object.defineProperty(window.document, 'cookie', {
    writable: true,
    value: 'access_token=FAKE_TOKEN_VAL',
  });

  const tree = renderer
    .create(
      <BrowserRouter>
        <AuthenticatedRoute>
          <></>
        </AuthenticatedRoute>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Redirects authenticated route page to login page if access token is not there', () => {
  Object.defineProperty(window.document, 'cookie', {
    writable: true,
    value: undefined,
  });

  const tree = renderer
    .create(
      <BrowserRouter>
        <Switch>
          <Route path="/login" exact component={Login} />
          <AuthenticatedRoute>
            <></>
          </AuthenticatedRoute>
        </Switch>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
