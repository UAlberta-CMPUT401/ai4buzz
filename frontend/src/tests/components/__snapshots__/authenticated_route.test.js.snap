// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redirects authenticated route page to login page if access token is not there 1`] = `
<div
  className="loginPage"
>
  <h1
    className="title"
  >
    AI4Buzz
  </h1>
  <form
    className="signinForm"
    onSubmit={[Function]}
  >
    <h1>
      Login
    </h1>
    <label
      className="labelName"
    >
      Email
    </label>
    <input
      className="signinInputEmail"
      onChange={[Function]}
      placeholder="email"
      type="email"
    />
    <label
      className="labelName"
    >
      Password
    </label>
    <input
      className="signinInputPassword"
      onChange={[Function]}
      placeholder="password"
      type="password"
    />
    <div
      className="buttonGradient"
    >
      <button
        className="submitButton"
      >
        Login
      </button>
    </div>
    <a
      className="switchAuth"
      href="/signup"
      onClick={[Function]}
    >
      Signup
    </a>
  </form>
</div>
`;

exports[`Renders authenticated route page correctly if access token is available 1`] = `null`;
